### ISSUE TITLE
Creating a singleplayer world crashes the client

### ISSUE THREAD

**Issue by NikBombadil** [McJty/RFToolsDimensions#252]
Issue: Creating a new singleplayer world crashes the client for some reason. Tried multiple times with the same results.

Versions:
•Minecraft: 1.12.2
•Forge: 14.23.5.2815
•McJtyLib: 3.1.1
•RFTools: 7.61
•RFToolsDimensions: 5.64

Crash Log: https://paste.dimdev.org/ogunenokuh.mccrash

**Comment by McJty** [comment]
Is this on a new world? That's very weird because it is trying to open an rftools dimension but on a new world there are none of those

**Comment by NikBombadil** [comment]
@McJty Yeah it's a completely fresh world, from a fresh install of a custom pack.

**Comment by McJty** [comment]
I don't understand how that's possible. I do however note several other errors in the log below. Can it be that somehow the pack is corrupted? Can you reproduce this with only rftools, rftools dimensions and mcjtylib?

**Comment by NikBombadil** [comment]
@McJty I've reloaded the pack with just those three mods and it works flawlessly. I guess it is either corrupted or a strange interaction. Thank you for your time! Would you have any advice on how to locate the culprit mods or would it be faster to just start a clean slate?

**Comment by McJty** [comment]
hmm hard to say really. Maybe starting clean might be easier

**Comment by NikBombadil** [comment]
@McJty Thanks again for your help! And for making such great mods, good luck!

**Comment by ghost** [comment]
A lot of my users have having this same exact crash when loading up existing worlds, unfortunately.

Edit: [Crashlog a user gave me](https://pastebin.com/bS2e9Jev)

**Comment by McJty** [comment]
'A lot of my users have having this same exact crash when loading up existing worlds, unfortunately.'? What do you mean exactly? When does this crash occur?

**Comment by Next-victim** [comment]
Similar crash happened to me wen I was doing a test world in 1.2.6

First run of a new world crashed to desktop, second run ran brought it up properly. I’ll check if I still have the crash log.

**Comment by Nawor3565** [comment]
Having the same issue. Can't create a new world or join an old one. When I create a new world, a folder called "RFTOOLS16" is made, but it's empty. 

Crash log: https://paste.dimdev.org/madobukaxe.mccrash
Forge: 14.23.5.2838
RFTools Dim: 1.12-5.71
RFTools: 1.12-7.72
McJtyLib: 1.12-3.5.4

Note: I run a dedicated server with the exact same mods, and it does not encounter this problem.

**Comment by ThatGamerBlue** [comment]
I'm getting this crash when switching from a multiplayer server with rftools dimensions to a singleplayer world without them, even on 5.71. Same stacktrace as everyone else.

**Comment by leytilera** [comment]
I have the same issue, seems like the thing that should be fixed in 5.71 but the 5.71 seems to changed nothing

**Comment by FederAndInk** [comment]
I had the same issue: 
[crash-2020-05-03_17.31.55-server.txt](https://github.com/McJtyMods/RFToolsDimensions/files/4570886/crash-2020-05-03_17.31.55-server.txt)

This was on a private modpack that I made with atlauncher and share it on the technic launcher.
The pack runs fine with atlauncher, no crash on singleplayer but with technic, there appears the crash.

After some diffs I found the culprit bin/minecraft.jar (technic) and libraries/net/minecraft/client/1.12.2/client-1.12.2.jar (atlauncher) were not the same (even if comparing decompressed jars were recursively identical).

Before I checked all the mods and configs and other files and libs: all were the same.

Finally, I tested by replacing the minecraft.jar of technic by the atlauncher one and no more crashes, yay :)

TL;DR: it seems that technic has a corrupted minecraft.jar in yourmodpackfolder/bin/minecraft.jar

Please tell us if you are using technic, and if replacing minecraft.jar fix the problem on your end.

**Comment by Nawor3565** [comment]
I have the problem using a private modpack that just uses the default minecraft.jar. Maybe you could compare the ATLauncher .jar to the vanilla one and see if they're identical?

**Comment by FederAndInk** [comment]
Yes, they are identical
here is the sha256sum: `8ada07da5ee77dad3527bd7278fbd05ee1fc8a597813b216a871a2d7d64cc64f`

**Comment by FederAndInk** [comment]
Also, I am using forge 1.12.2-14.23.5.2847

**Comment by corgiblu** [comment]
Popping in to say I'm having this issue on the latest version of MC Eternal. I'm away from my PC at the moment so I can't get the versions of RFTools. I'm able to join a multiplayer server, but if I try to join (or create a new) single player world it crashes.
